protocol_decoder/src/decoding.rs

```diff
@@ -317,9 +322,23 @@ impl ProcessedBlockTrace {
                 true
             }
             1 => {
-                // Just need one.
-                let dummy_txn = create_dummy_gen_input(other_data, extra_data, initial_tries);
-                gen_inputs.insert(0, dummy_txn);
+                // We just need one dummy entry.
+                // If there are withdrawals, we will need to append them at the end of the block
+                // execution, in which case we directly append the dummy proof
+                // after the only txn of this block.
+                // If there are no withdrawals, then the dummy proof will be prepended to the
+                // actual txn.
+                match has_withdrawals {
```

Comment:
Why not use `if { … } else { … }`?

Reply:
I actually find that using a `match` on a `bool` is usually more readable (there might be something wrong with me, idk :upside_down_face:) than an `if / else`. I think using a match is a bit less common though.

Reply:
Yeah not sure why I went for match, maybe an implicit preference as Brendan said, happy to change though

Reply:
I'd vote for keeping match